app/src/rgb_underglow.c

```diff
@@ -265,7 +271,13 @@ static int zmk_rgb_underglow_init(const struct device *_arg) {
     settings_load_subtree("rgb/underglow");
 #endif
 
+#if IS_ENABLED(CONFIG_ZMK_RGB_UNDERGLOW_AUTO_OFF_USB)
+    state.on = zmk_usb_is_powered();
+    if (state.on)
+        k_timer_start(&underglow_tick, K_NO_WAIT, K_MSEC(50));
+#else
     k_timer_start(&underglow_tick, K_NO_WAIT, K_MSEC(50));
+#endif
```

Comment:
Can be simplified, I believe:

```suggestion
#if IS_ENABLED(CONFIG_ZMK_RGB_UNDERGLOW_AUTO_OFF_USB)
    state.on = zmk_usb_is_powered();
#endif

    if (state.on) {
        k_timer_start(&underglow_tick, K_NO_WAIT, K_MSEC(50));
    }
```

Reply:
I didnt want to do that bc i wasn't sure if the timer had to be running for other stuff and didn't want to interfere but if you're ok with it then that is definitely the better way

Reply:
This was sorta a bug, because the timer was running even if the leds are off on init, so probably goodto fix that, saves a bit of power

Reply:
Yeah, that was my thinking as well. Thanks!